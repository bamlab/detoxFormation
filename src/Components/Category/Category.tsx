import React, { FunctionComponent } from 'react';
import { StyleSheet, Text, View, Image } from 'react-native';

type Category = { strCategory: string; strCategoryThumb: string };

interface CategoryProps {
  category: Category;
}

export const Category: FunctionComponent<CategoryProps> = ({
  category: { strCategory, strCategoryThumb },
}) => {
  return (
    <View style={styles.category}>
      <Text style={styles.text}>{strCategory}</Text>
      <Image style={styles.img} source={{ uri: strCategoryThumb }} />
    </View>
  );
};

const styles = StyleSheet.create({
  category: {
    flexDirection: 'row',
    alignItems: 'center',
    margin: 10,
  },
  text: {
    fontSize: 20,
    marginLeft: 20,
    color: 'black',
    backgroundColor: '#FFF8',
    position: 'absolute',
    zIndex: 1,
  },
  img: {
    width: '100%',
    height: 75,
  },
});
