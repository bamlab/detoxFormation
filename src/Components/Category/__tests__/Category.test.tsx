import 'react-native';
import React from 'react';
import { render } from '@testing-library/react-native';
import { Category } from '../Category';

describe('[Component] Category', () => {
  test('renders correctly', () => {
    const tree = render(
      <Category
        category={{
          strCategory: 'Category A',
          strCategoryThumb: 'Link to the category A image',
        }}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
