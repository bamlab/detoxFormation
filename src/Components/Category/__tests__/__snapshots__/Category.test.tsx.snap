// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Category renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "margin": 10,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#FFF8",
        "color": "black",
        "fontSize": 20,
        "marginLeft": 20,
        "position": "absolute",
        "zIndex": 1,
      }
    }
  >
    Category A
  </Text>
  <Image
    source={
      Object {
        "uri": "Link to the category A image",
      }
    }
    style={
      Object {
        "height": 75,
        "width": "100%",
      }
    }
  />
</View>
`;
